test/rules/no-duplicate-super/test.ts.lint

```diff
@@ -0,0 +1,40 @@
+class A {
+    constructor() {
+        super();
+        super();
+        ~~~~~ [0]
+    }
+}
+
+class B {
+    constructor() {
+        super();
+        super.foo();
+    }
+}
+
+class C {
+    constructor() {
+        super();
+
+        class C2 {
+            constructor() {
+                super();
+            }
+        }
+    }
+}
+
+class D {
+    // Unfortunately, we warn here too.
+    constructor() {
+        if (cond()) {
+            super(true);
+        } else {
+            super(false);
+            ~~~~~ [0]
```

Comment:
What if the `if` has a `return` instead of `else`? What if there's a `switch` statement with `break`?
TypeScript does control flow analysis, but as far as I can tell, there's no API for accessing it. But I would hate to add a bunch of exceptions that duplicate work done better by TypeScript itself.